scripts/zones/Tahrongi_Canyon/Zone.lua

```diff
@@ -132,4 +133,19 @@ function onEventFinish( player, csid, option)
     if (csid == 35) then
         lightCutsceneFinish(player); -- Quest: I Can Hear A Rainbow
     end
+end;
+
+function isHabrokWeather(weather)
+    if (weather == WEATHER_DUST_STORM or weather == WEATHER_SAND_STORM or weather == WEATHER_WIND or weather == WEATHER_GALES) then
+        return true;
+    end
+    return false;
+end
+
+function onZoneWeatherChange(weather)
+    if (GetMobByID(HABROK):isSpawned() and not isHabrokWeather(weather)) then
```

Comment:
think we might wanna stick the return value of GetMobByID in a local variable at the top of this function instead of looking up the mob each time

Reply:
Done